build_tools/travis/test_script.sh

```diff
@@ -33,3 +33,11 @@ if [[ "$BUILD_DOCS" == "true" ]]; then
   make html-strict
   cd ..
 fi
+
+if [[ "$COVERAGE" == "true" ]]; then
```

Comment:
@matt-gardner @DeNeutoy how important do you think it was to have `codecov` in after_success?  I'm not sure how to do this if we use Docker builds.

Reply:
I could probably maintain this by keeping the container running an not using `--rm`, but I think it will end up taking a bit of iteration.